torch/testing/_internal/common_distributed.py

```diff
@@ -878,6 +889,77 @@ def _check_return_codes(self, elapsed_time) -> None:
     def is_master(self) -> bool:
         return self.rank == 0
 
+# Utility base class for distributed Multi Process Test cases
+# This abstracts the PG creation and deletion, the backends are selected based
+# on device type. The tests functions can be instantiated per device type using
+# common_device_type.instantiate_device_type_tests
+# other backends can add entry in backend() function
+class DistributedTestBase(MultiProcessTestCase):
+
+    def setUp(self):
+        super().setUp()
+        self._spawn_processes()
+
+    def tearDown(self):
+        try:
+            os.remove(self.file_name)
+        except OSError:
+            pass
+
+    def backend(self, device) -> str:
+        if "cuda" in device:
+            return "nccl"
+        elif "hpu" in device :   # intel gaudi
+            return "hccl"
+        else :
+            return "gloo"
+
+    def device_handle(self, device) :
```

Comment:
I see torch.get_device_module API available with this commit : https://github.com/pytorch/pytorch/commit/73744a2c009b34d818de8b1835fae75808bf58ee#diff-c8835eba8562819a44e70a30545213c96e68c588a2cfa673fa495b187b1ea8dfR2021

 maybe we can refine this further as part of a subsequent PR update after ensuring it has no impact on out-of-tree devices like hpu.

Reply:
addressed in: 04ada419407cd75565afaa7748817c2a4454ecfe